edxval/utils.py

```diff
@@ -243,3 +245,41 @@ def validate_generated_images(value, max_items):
         raise ValidationError(u'list must only contain strings.')
 
     return value
+
+
+def generate_file_content_hash(uploaded_file):
+    """
+    Generates SHA256 Content Hash for a File
+
+    Arguments:
+        uploaded_file (UploadedFile): File which will be used for hash generation
+
+    Returns:
+        str sha256 hash
+    """
+    content_hash = hashlib.sha256()
+
+    with closing(uploaded_file.open()) as file_content:
+        for chunk in iter(lambda: file_content.read(65536), b""):
```

Comment:
Out of curiosity: Is there any bytes limitation on iter? I am interested about the choice of 65536 when reading the content. How will it behave if the contents to be read have bytes less than 65536?

Reply:
> Is there any bytes limitation on iter? I am interested about the choice of 65536 when reading the content.

No, not that I know of. 65536 bytes is about 64 kilobytes. I specified that size in order to avoid allocating more memory to reading the transcript.

> How will it behave if the contents to be read have bytes less than 65536?

It would behave similarly to looping over an array with a single item. 

---

I changed this part to `file_content.read()` directly, instead, disregarding the memory constraints. 

Please let me know if you'd like me to revert that change :+1: